### PR TITLE
adjustments to propagate keep_metrics argument

### PR DESCRIPTION
**Pull request by snk-actian**
`keep-metrics` argument is defined with default value and used in following files:

- relabelings/kubelet.river
- relabelings/kubelet-cadvisor.river
- relabelings/kube-state-metrics.river
- relabelings/node-exporter.river

But this argument is not defined in caller modules to update based on use-case.

This PR will allow one to define custom value for `keep-metrics` while calling `metrics/all.river` in grafana-agent configuration file and it will be propagated down to modules in relabelings folder.

### CODE REVIEW

**Review by erikbaranowski — APPROVED** [review]
LGTM but I'll give @bentonam a little time to get eyes on before I merge.

File: modules/kubernetes/metrics/all.river
Comment (erikbaranowski) [inline modules/kubernetes/metrics/all.river:62]: should the default be set here same as /modules/kubernetes/metrics/relabelings/kube-apiserver.river? Similar question for keep_metrics default and for both as they get passed down.

```default = "kubelet_(pod_worker_latency_microseconds|pod_start_latency_microseconds|cgroup_manager_latency_microseconds|pod_worker_start_latency_microseconds|pleg_relist_latency_microseconds|pleg_relist_interval_microseconds|runtime_operations|runtime_operations_latency_microseconds|runtime_operations_errors|eviction_stats_age_microseconds|device_plugin_registration_count|device_plugin_alloc_latency_microseconds|network_plugin_operations_latency_microseconds)|scheduler_(e2e_scheduling_latency_microseconds|scheduling_algorithm_predicate_evaluation|scheduling_algorithm_priority_evaluation|scheduling_algorithm_preemption_evaluation|scheduling_algorithm_latency_microseconds|binding_latency_microseconds|scheduling_latency_seconds)|apiserver_(request_count|request_latencies|request_latencies_summary|dropped_requests|storage_data_key_generation_latencies_microseconds|storage_transformation_failures_total|storage_transformation_latencies_microseconds|proxy_tunnel_sync_latency_secs|longrunning_gauge|registered_watchers)|kubelet_docker_(operations|operations_latency_microseconds|operations_errors|operations_timeout)|reflector_(items_per_list|items_per_watch|list_duration_seconds|lists_total|short_watches_total|watch_duration_seconds|watches_total)|etcd_(helper_cache_hit_count|helper_cache_miss_count|helper_cache_entry_count|object_counts|request_cache_get_latencies_summary|request_cache_add_latencies_summary|request_latencies_summary)|transformation_(transformation_latencies_microseconds|failures_total)|(admission_quota_controller_adds|admission_quota_controller_depth|admission_quota_controller_longest_running_processor_microseconds|admission_quota_controller_queue_latency|admission_quota_controller_unfinished_work_seconds|admission_quota_controller_work_duration|APIServiceOpenAPIAggregationControllerQueue1_adds|APIServiceOpenAPIAggregationControllerQueue1_depth|APIServiceOpenAPIAggregationControllerQueue1_longest_running_processor_microseconds|APIServiceOpenAPIAggregationControllerQueue1_queue_latency|APIServiceOpenAPIAggregationControllerQueue1_retries|APIServiceOpenAPIAggregationControllerQueue1_unfinished_work_seconds|APIServiceOpenAPIAggregationControllerQueue1_work_duration|APIServiceRegistrationController_adds|APIServiceRegistrationController_depth|APIServiceRegistrationController_longest_running_processor_microseconds|APIServiceRegistrationController_queue_latency|APIServiceRegistrationController_retries|APIServiceRegistrationController_unfinished_work_seconds|APIServiceRegistrationController_work_duration|autoregister_adds|autoregister_depth|autoregister_longest_running_processor_microseconds|autoregister_queue_latency|autoregister_retries|autoregister_unfinished_work_seconds|autoregister_work_duration|AvailableConditionController_adds|AvailableConditionController_depth|AvailableConditionController_longest_running_processor_microseconds|AvailableConditionController_queue_latency|AvailableConditionController_retries|AvailableConditionController_unfinished_work_seconds|AvailableConditionController_work_duration|crd_autoregistration_controller_adds|crd_autoregistration_controller_depth|crd_autoregistration_controller_longest_running_processor_microseconds|crd_autoregistration_controller_queue_latency|crd_autoregistration_controller_retries|crd_autoregistration_controller_unfinished_work_seconds|crd_autoregistration_controller_work_duration|crdEstablishing_adds|crdEstablishing_depth|crdEstablishing_longest_running_processor_microseconds|crdEstablishing_queue_latency|crdEstablishing_retries|crdEstablishing_unfinished_work_seconds|crdEstablishing_work_duration|crd_finalizer_adds|crd_finalizer_depth|crd_finalizer_longest_running_processor_microseconds|crd_finalizer_queue_latency|crd_finalizer_retries|crd_finalizer_unfinished_work_seconds|crd_finalizer_work_duration|crd_naming_condition_controller_adds|crd_naming_condition_controller_depth|crd_naming_condition_controller_longest_running_processor_microseconds|crd_naming_condition_controller_queue_latency|crd_naming_condition_controller_retries|crd_naming_condition_controller_unfinished_work_seconds|crd_naming_condition_controller_work_duration|crd_openapi_controller_adds|crd_openapi_controller_depth|crd_openapi_controller_longest_running_processor_microseconds|crd_openapi_controller_queue_latency|crd_openapi_controller_retries|crd_openapi_controller_unfinished_work_seconds|crd_openapi_controller_work_duration|DiscoveryController_adds|DiscoveryController_depth|DiscoveryController_longest_running_processor_microseconds|DiscoveryController_queue_latency|DiscoveryController_retries|DiscoveryController_unfinished_work_seconds|DiscoveryController_work_duration|kubeproxy_sync_proxy_rules_latency_microseconds|non_structural_schema_condition_controller_adds|non_structural_schema_condition_controller_depth|non_structural_schema_condition_controller_longest_running_processor_microseconds|non_structural_schema_condition_controller_queue_latency|non_structural_schema_condition_controller_retries|non_structural_schema_condition_controller_unfinished_work_seconds|non_structural_schema_condition_controller_work_duration|rest_client_request_latency_seconds|storage_operation_errors_total|storage_operation_status_count)|etcd_(debugging|disk|server).*|apiserver_admission_controller_admission_latencies_seconds_.*|apiserver_admission_step_admission_latencies_seconds_.*"```

Comment (snk-actian) [reply modules/kubernetes/metrics/all.river:62]: These default values kind of duplicating in a couple of relabilings .river files. Could not find proper time to test those. May be @bentonam can give some feedback. 🙏🏻

**Comment by snk-actian** [comment]
> WIP: Found some glitches.. Trying to fix.

With latest commit, it works with my test clusters..

**Comment by snk-actian** [comment]
@bentonam @erikbaranowski Can you please re-review?

**Comment by bentonam** [comment]
@snk-actian Thank you for the contribution.  I had not exposed that initially in the `all.river` as I felt like it was getting very argument heavy and didn't want to end up with dozens of options, as a module should be fairly straight forward.   I believe the regex is the same as the K8s integration in Grafana Cloud extracted all of the metrics referenced in dashboards/rules for the integration and not on a per job basis.  My thought by exposing the argument in the relabelings was to allow a user to directly reference that specific module in their config.  

I do have a PR that changes the behavior of `modules/kubernetes/logs/all.river` and `modules/kubernetes/logs/all.river` to no longer accept credentials but a `list(LogsReceiver)` or `list(MetricsReceiver)` and reduces the overall number of arguments passed to the module https://github.com/grafana/agent-modules/pull/27

I would like to at least wait until that PR is merged, as it will create conflicts for this PR.  But I want to evaluate it a little more, as we need to balance between adding more and more arguments vs. users consuming modules directly and forwarding to their own receivers to change the behavior.   Take a look at some of the examples in that PR: 

https://github.com/grafana/agent-modules/blob/k8s-annotations-writer-arg/example/kubernetes/metrics/selective-modules.river
https://github.com/grafana/agent-modules/blob/k8s-annotations-writer-arg/example/kubernetes/metrics/multi-target-example.river
https://github.com/grafana/agent-modules/blob/k8s-annotations-writer-arg/example/kubernetes/metrics/custom-rewrites.river

**Comment by bentonam** [comment]
I am going to close this PR as it is stale and for the reasons discussed above.   Also new modules have been added with independent jobs that can be consumed separately, reducing complexity.  

https://github.com/grafana/agent-modules/tree/main/modules/kubernetes/metrics/jobs

**Pull request closed without merging**
